### PR TITLE
refactor(toc_helper): utilize hexo-util

### PR DESCRIPTION
**Pull request by SukkaW**
## What does it do?

This PR is the last pieces of #3677. We finally made it.

## How to test

```sh
git clone -b htmlparser2-toc https://github.com/sukkaw/hexo.git
cd hexo
npm install
npm test
```

## Screenshots

Since `hexo-theme-landscape` is not using toc helper, I have set up a benchmark using my `hexo-theme-suka`, with hexo's `meta_generator` and `external_link` disabled.

**Node.js 8**

| | Process Time | Memory  |
|:---:|:---:|:---:|
| Current Master Branch | 21s | 528MB |
| This PR | 17s | 421MB |

**Nodejs.10**

| | Process Time | Memory  |
|:---:|:---:|:---:|
| Current Master Branch | 20s | 527MB |
| This PR | 17s | 464MB |

**Nodejs.12**

| | Process Time | Memory  |
|:---:|:---:|:---:|
| Current Master Branch | 24s | 460MB |
| This PR | 22s | 470MB |

## Pull request tasks

- [ ] Add test cases for the changes.
- [x] Passed the CI test.

### CODE REVIEW

**Comment by SukkaW** [comment]
After this PR is merged, I might begin to address #3288.

**Comment by SukkaW** [comment]
This PR should be superseded by utilizing newer version of `hexo-util` after https://github.com/hexojs/hexo-util/pull/137 is merged and released in `hexo-util@1.6.0`.

**Comment by curbengh** [comment]
> This PR should be superseded by utilizing newer version of `hexo-util` after [hexojs/hexo-util#137](https://github.com/hexojs/hexo-util/pull/137).

Can this PR use https://github.com/hexojs/hexo-util/pull/137 branch temporarily for testing? Once that PR is released, just need to update hexo-util in this PR.

File: lib/plugins/helper/toc.js [hunk 1]
```diff
@@ -1,35 +1,29 @@
...
-const { escapeHTML } = require('hexo-util');
+const { tocObj } = require('hexo-util');
 
 function tocHelper(str, options = {}) {
-  if (!cheerio) cheerio = require('cheerio');
+  options = Object.assign({
+    max_depth: 6,
+    class: 'toc',
+    list_number: true
+  }, options);
```
Comment (curbengh) [inline lib/plugins/helper/toc.js]: `min_depth` option. I don't mind if you prefer to add that option in a separate PR.

Comment (SukkaW) [reply lib/plugins/helper/toc.js]: Maybe in this PR then since `tocObj` already has `min_depth` bring up.

Comment (SukkaW) [reply lib/plugins/helper/toc.js]: @curbengh No. The current unit test for toc helper is completely a mess. I might refactor the unit test when bringing up the `min_depth`.

**Comment by curbengh** [comment]
[So far](https://github.com/hexojs/hexo/compare/4.1.1...master) we haven't merge any new feature yet, perhaps this can be released into 4.1.2?

**Comment by SukkaW** [comment]
@curbengh `min_depth` could be bring up along with `tocObj()` so there won't be only one feature.

**Comment by SukkaW** [comment]
@curbengh Rebased.

**Comment by curbengh** [comment]
> `min_depth` could be bring up along with `tocObj()` so there won't be only one feature.

That makes two features then. Sufficient for 4.2.0. #3891 can be deferred to 4.3+.

**Pull request merged**